### ISSUE TITLE
Default Snowflake JDBC driver throws exception immediately when running queries in 23.3.2

### ISSUE THREAD

**Issue by hlindqvist** [dbeaver/dbeaver#22502]
### Description

When running user queries, the execution fails because of an exception in the Snowflake JDBC driver.
This seems to be because of some incompatibility between the predefined Snowflake driver version shipped with DBeaver and DBeaver itself as of 23.3.2.
```
org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [200001] [XX000]: JDBC driver internal error: Fail to retrieve row count for first arrow chunk: null.
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:133)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:600)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$2(SQLQueryJob.java:503)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:510)
	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:962)
	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:4130)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:123)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:190)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:121)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5148)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: net.snowflake.client.jdbc.SnowflakeSQLLoggedException: JDBC driver internal error: Fail to retrieve row count for first arrow chunk: null.
	at net.snowflake.client.jdbc.SnowflakeResultSetSerializableV1.setFirstChunkRowCountForArrow(SnowflakeResultSetSerializableV1.java:1011)
	at net.snowflake.client.jdbc.SnowflakeResultSetSerializableV1.create(SnowflakeResultSetSerializableV1.java:504)
	at net.snowflake.client.core.SFResultSetFactory.getResultSet(SFResultSetFactory.java:28)
	at net.snowflake.client.core.SFStatement.executeQueryInternal(SFStatement.java:251)
	at net.snowflake.client.core.SFStatement.executeQuery(SFStatement.java:167)
	at net.snowflake.client.core.SFStatement.execute(SFStatement.java:749)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.executeInternal(SnowflakeStatementV1.java:292)
	at net.snowflake.client.jdbc.SnowflakeStatementV1.execute(SnowflakeStatementV1.java:342)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:330)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	... 12 more
	```

### DBeaver Version

Community Edition 23.3.2

### Operating System

Windows 10

### Database and driver

Snowflake with driver version 3.13.30 as is predefined in DBeaver 23.3.2

Note: Editing the Snowflake settings in Driver Manager in DBeaver and replacing the Snowflake JDBC driver with version 3.14.4 (current as of right now) seems to address the problem

### Steps to reproduce

1. Have DBeaver set up with a Snowflake connection using the predefined driver for Snowflake as-is
2. Run any user query (as opposed to just expanding the tree etc), ie open an editor and run eg `SELECT * FROM mytable`, or even choose "View Data" on a table in the tree.
  

### Additional context

Snowflake JDBC driver version 3.14.4 does appear to work correctly in 23.3.2, it is just not the "prepackaged" version that DBeaver ships

**Comment by LonwoLonwo** [comment]
Hello @hlindqvist 
Do you use Windows Store or a standalone version?

**Comment by hlindqvist** [comment]
> Hello @hlindqvist Do you use Windows Store or a standalone version?

Ah, sorry yes. This is 23.3.2 installed from Windows Store

**Comment by LonwoLonwo** [comment]
Thanks for the report.

**Comment by LonwoLonwo** [comment]
The whole discussion was here: https://github.com/dbeaver/dbeaver/issues/14691.

As a workaround for now - you can avoid this issue with this bootstrap query:

`ALTER SESSION SET JDBC_QUERY_RESULT_FORMAT='JSON'`

![2024-01-18 17_59_49-Connection _TEST_DB 1_ configuration](https://github.com/dbeaver/dbeaver/assets/45152336/4bc35c56-cb2d-4925-b364-5ccdee3f9d0f)

**Comment by hlindqvist** [comment]
> The whole discussion was here: #14691.
> 
> As a workaround for now - you can avoid this issue with this bootstrap query:
> 
> `ALTER SESSION SET JDBC_QUERY_RESULT_FORMAT='JSON'`
> 


Alright, interesting background on what is actually wrong.

Also, just to make it completely clear, the following workaround (also mentioned in my initial report) seems to work fine as well, and seems like something that would probably be more suited to ship as the new default in Dbeaver?


![image (4)](https://github.com/dbeaver/dbeaver/assets/1497583/397990d1-80e5-47c0-abcb-8695e4dc3d43)

**Comment by LonwoLonwo** [comment]
> Snowflake JDBC driver version 3.14.4 does appear to work correctly in 23.3.2

This may be a correct decision. Thanks.

**Comment by LonwoLonwo** [comment]
Fixed. The fix will be available in the next DBeaver version, 23.3.4, on 05-02-2024.
